Tests/UITests/AlertPresentableSpec.swift

```diff
@@ -0,0 +1,85 @@
+import XCTest
+import Quick
+import Nimble
+import class Shock.MockServer
+
+class AlertPresentableSpec: QuickSpec {
+
+    override func spec() {
+
+        var mockServer: MockServer!
+        var app: XCUIApplication!
+
+        func launchAppIfNecessary(context: String) {
+            mockServer.setup(route: MockServerHelper.pingRouteMock(jsonStub: context))
+            mockServer.start()
+
+            guard app == nil else {
+                return
+            }
+            app = XCUIApplication()
+            app.launchArguments.append("--uitesting")
+            app.launchArguments.append("-context \(context)")
+            app.launch()
+        }
+
+        beforeEach {
+            mockServer = MockServerHelper.setupNewServer(route: MockServerHelper.standardRouting)
+        }
+
+        afterEach {
+            mockServer.stop()
+            app = nil
+        }
+
+        describe("AlertPresentable") {
+
+            context("with modal view") {
+
+                var iamView: XCUIElement {
+                    app.otherElements.element(matching: NSPredicate(format: "identifier BEGINSWITH[cd] 'IAMView'"))
+                }
+
+                beforeEach {
+                    launchAppIfNecessary(context: "modal-controls-invalid-url")
+                    if !iamView.exists {
+                        app.buttons["purchase_successful"].tap()
+                        expect(iamView.exists).toEventually(beTrue(), timeout: .seconds(2))
+                    }
+                }
+
+                it("should show error alert after tapping button 1 and close campaign") {
+                    iamView.buttons["Button0"].tap()
+                    expect(app.alerts.element.staticTexts["Page not found"].exists).to(beTrue())
+                    app.alerts.element.buttons["Close"].tap()
+                }
+            }
+
+            context("with slide up view") {
+
+                var iamView: XCUIElement {
+                    app.otherElements["IAMView-SlideUp"]
+                }
+                var content: XCUIElement {
+                    iamView.buttons["bodyMessage"]
+                }
+
+                beforeEach {
+                    launchAppIfNecessary(context: "slide-up-trigger-invalid-url")
+                    if !iamView.exists {
+                        app.buttons["purchase_successful"].tap()
+                        expect(iamView.exists).toEventually(beTrue(), timeout: .seconds(2))
+                    }
+                }
+
+                it("should show error alert after tapping button 1 and close campaign") {
+                    content.tap()
+                    expect(app.alerts.element.staticTexts["Page not found"].exists).to(beTrue())
+                    app.alerts.element.buttons["Close"].tap()
+                    iamView.buttons["exitButton"].tap()
```

Comment:
tapping the content doesn't close the campaign? (Are those two lines necessary?)

Reply:
It is not necessary, removed here [00a6b6a](https://github.com/rakutentech/ios-inappmessaging/pull/269/commits/00a6b6aaefe88e6b43e71bb8da6dc3997aa1f3f8).